src/serialize.h

```diff
@@ -411,6 +422,31 @@ class CVarInt
     }
 };
 
+template<typename I>
+class BigEndian
+{
+protected:
+    I& m_val;
+public:
+    explicit BigEndian(I& val) : m_val(val)
+    {
+        static_assert(sizeof(I) == 2, "Unsupported BigEndian size");
+        static_assert(std::is_unsigned<I>::value, "BigEndian type must be unsigned integer");
+    }
+
+    template<typename Stream>
+    void Serialize(Stream& s) const
+    {
+        if (sizeof(I) == 2) ser_writedata16be(s, m_val);
```

Comment:
Why these checks `if (sizeof(I) == 2)`?

Reply:
To more easily extend this later (if needed) to other sizes. I can get rid of them if that's confusing.

Reply:
At the moment is confusing/redundant because of the static assert above. Keep it simple?

Reply:
Fixed.